src/main/java/dev/openfga/sdk/api/configuration/ConfigurationOverride.java

```diff
@@ -179,4 +182,28 @@ public ConfigurationOverride minimumRetryDelay(Duration minimumRetryDelay) {
     public Duration getMinimumRetryDelay() {
         return minimumRetryDelay;
     }
+
+    public ConfigurationOverride additionalHeaders(Map<String, String> additionalHeaders) {
+        this.additionalHeaders = additionalHeaders;
+        return this;
+    }
+
+    public ConfigurationOverride addHeaders(AdditionalHeadersSupplier supplier) {
+        if (supplier == null || supplier.getAdditionalHeaders() == null) {
+            // No headers to add.
+            return this;
+        }
+
+        var additionalHeaders = supplier.getAdditionalHeaders();
```

Comment:
Just to make sure what should happen is that the default headers in the Config should be there, additional headers per request update them.

So if the same header is overridden we use the one from the override, but if a particular header is not overriden it is passed in.

To remove a particular header from a request, you'd override it with value null

Reply:
This is the behavior with the overrides. I'll add tests to make this behavior both more explicit and also to prevent regressions

Reply:
added tests (and removed a related typing issue) in https://github.com/openfga/java-sdk/pull/37/commits/23603940828703435405356892b57c14add3cfcb